### ISSUE TITLE
[blob] PageBlobClient.getPageRangesDiff() returning empty clearRange when prevSnapshot is specified

### ISSUE THREAD

**Issue by ljian3377** [Azure/Azurite#298]
### Which service(blob, file, queue, table) does this issue concern?
blob
### Which version of the Azurite was used?
3.3.0-preview
### Where do you get Azurite? (npm, DockerHub, NuGet, Visual Studio Code Extension)
npm
### What's the Node.js version?
v12.13.0
### What problem was encountered?
Unit test failure:
PageBlobClient: getPageRangesDiff
```
  it.only("getPageRangesDiff", async () => {
    await pageBlobClient.create(1024);

    const result = await blobClient.download(0);
    assert.deepStrictEqual(await bodyToString(result, 1024), "\u0000".repeat(1024));

    await pageBlobClient.uploadPages("b".repeat(1024), 0, 1024);

    const snapshotResult = await pageBlobClient.createSnapshot();
    assert.ok(snapshotResult.snapshot);

    await pageBlobClient.uploadPages("a".repeat(512), 0, 512);
    await pageBlobClient.clearPages(512, 512);

    const rangesDiff = await pageBlobClient.getPageRangesDiff(0, 1024, snapshotResult.snapshot!);

    assert.equal(rangesDiff.pageRange![0].offset, 0);
    assert.equal(rangesDiff.pageRange![0].count, 511);
    assert.equal(rangesDiff.clearRange![0].offset, 512);
    assert.equal(rangesDiff.clearRange![0].count, 511);
  });
```
Assert failed with ` assert.equal(rangesDiff.clearRange![0].offset, 512)`, as clearRange is an empty array. 

```
export interface PageList {
...
  /**
   * Present if the prevSnapshot parameter was specified and there were cleared
   * pages between the previous snapshot and the target snapshot.
   */
  clearRange?: Range[];
}

```

### Steps to reproduce the issue?
Re-run the test-case.

### Have you found a mitigation/solution?
N/A

### debug.log
```
2019-12-13T07:46:30.272Z 79f574d0-5f3e-4a85-ac0f-c607b37fb1b1 info: BlobStorageContextMiddleware: RequestMethod=GET RequestURL=http://127.0.0.1/devstoreaccount1/container157622319013805656/blob157622319019106382?prevsnapshot=2019-12-13T07%3A46%3A30.235Z&comp=pagelist RequestHeaders:{"x-ms-range":"bytes=0-1023","x-ms-version":"2019-02-02","user-agent":"azsdk-js-storageblob/12.0.1 (NODE-VERSION v12.13.0; Windows_NT 10.0.18362)","x-ms-client-request-id":"a434b9dd-84cb-49aa-b6ce-0ed0ce8482c1","x-ms-date":"Fri, 13 Dec 2019 07:46:30 GMT","authorization":"SharedKey devstoreaccount1:N0VcwTcnm161yNzBV+f4SvHgXDZrLGs/6oQ7HdE8EFk=","cookie":"","accept":"*/*","accept-encoding":"gzip,deflate","host":"127.0.0.1:10000","connection":"keep-alive"} ClientIP=127.0.0.1 Protocol=http HTTPVersion=1.1
2019-12-13T07:46:30.272Z 79f574d0-5f3e-4a85-ac0f-c607b37fb1b1 info: BlobStorageContextMiddleware: Account=devstoreaccount1 Container=container157622319013805656 Blob=blob157622319019106382
2019-12-13T07:46:30.272Z 79f574d0-5f3e-4a85-ac0f-c607b37fb1b1 verbose: DispatchMiddleware: Dispatching request...
2019-12-13T07:46:30.272Z 79f574d0-5f3e-4a85-ac0f-c607b37fb1b1 info: DispatchMiddleware: Operation=PageBlob_GetPageRanges
2019-12-13T07:46:30.272Z 79f574d0-5f3e-4a85-ac0f-c607b37fb1b1 verbose: AuthenticationMiddlewareFactory:createAuthenticationMiddleware() Validating authentications.
2019-12-13T07:46:30.272Z 79f574d0-5f3e-4a85-ac0f-c607b37fb1b1 info: PublicAccessAuthenticator:validate() Start validation against public access.
2019-12-13T07:46:30.272Z 79f574d0-5f3e-4a85-ac0f-c607b37fb1b1 debug: PublicAccessAuthenticator:validate() Getting account properties...
2019-12-13T07:46:30.272Z 79f574d0-5f3e-4a85-ac0f-c607b37fb1b1 debug: PublicAccessAuthenticator:validate() Retrieved account name from context: devstoreaccount1, container: container157622319013805656, blob: blob157622319019106382
2019-12-13T07:46:30.273Z 79f574d0-5f3e-4a85-ac0f-c607b37fb1b1 debug: PublicAccessAuthenticator:validate() Skip public access authentication. Cannot get public access type for container container157622319013805656
2019-12-13T07:46:30.273Z 79f574d0-5f3e-4a85-ac0f-c607b37fb1b1 info: BlobSharedKeyAuthenticator:validate() Start validation against account shared key authentication.
2019-12-13T07:46:30.273Z 79f574d0-5f3e-4a85-ac0f-c607b37fb1b1 info: BlobSharedKeyAuthenticator:validate() [STRING TO SIGN]:"GET\n\n\n\n\n\n\n\n\n\n\n\nx-ms-client-request-id:a434b9dd-84cb-49aa-b6ce-0ed0ce8482c1\nx-ms-date:Fri, 13 Dec 2019 07:46:30 GMT\nx-ms-range:bytes=0-1023\nx-ms-version:2019-02-02\n/devstoreaccount1/devstoreaccount1/container157622319013805656/blob157622319019106382\ncomp:pagelist\nprevsnapshot:2019-12-13T07:46:30.235Z"
2019-12-13T07:46:30.273Z 79f574d0-5f3e-4a85-ac0f-c607b37fb1b1 info: BlobSharedKeyAuthenticator:validate() Calculated authentication header based on key1: SharedKey devstoreaccount1:N0VcwTcnm161yNzBV+f4SvHgXDZrLGs/6oQ7HdE8EFk=
2019-12-13T07:46:30.273Z 79f574d0-5f3e-4a85-ac0f-c607b37fb1b1 info: BlobSharedKeyAuthenticator:validate() Signature 1 matched.
2019-12-13T07:46:30.273Z 79f574d0-5f3e-4a85-ac0f-c607b37fb1b1 verbose: DeserializerMiddleware: Start deserializing...
2019-12-13T07:46:30.273Z 79f574d0-5f3e-4a85-ac0f-c607b37fb1b1 info: HandlerMiddleware: DeserializedParameters={"options":{"range":"bytes=0-1023","requestId":"a434b9dd-84cb-49aa-b6ce-0ed0ce8482c1","leaseAccessConditions":{},"modifiedAccessConditions":{}},"comp":"pagelist","version":"2019-02-02"}
2019-12-13T07:46:30.273Z 79f574d0-5f3e-4a85-ac0f-c607b37fb1b1 verbose: SerializerMiddleware: Start serializing...
2019-12-13T07:46:30.275Z 79f574d0-5f3e-4a85-ac0f-c607b37fb1b1 debug: Serializer: Raw response body string is <?xml version="1.0" encoding="UTF-8" standalone="yes"?><PageList><PageRange><Start>0</Start><End>511</End></PageRange></PageList>
2019-12-13T07:46:30.275Z 79f574d0-5f3e-4a85-ac0f-c607b37fb1b1 info: Serializer: Start returning stream body.
2019-12-13T07:46:30.275Z 79f574d0-5f3e-4a85-ac0f-c607b37fb1b1 info: EndMiddleware: End response. TotalTimeInMS=3 StatusCode=200 StatusMessage=OK Headers={"server":"Azurite-Blob/3.3.0-preview","last-modified":"Fri, 13 Dec 2019 07:46:30 GMT","etag":"\"d-B4b5WyEWP84Dg5D6mWYwVnuQ/YE\"","x-ms-blob-content-length":"1024","x-ms-client-request-id":"a434b9dd-84cb-49aa-b6ce-0ed0ce8482c1","x-ms-request-id":"79f574d0-5f3e-4a85-ac0f-c607b37fb1b1","x-ms-version":"2019-02-02","date":"Fri, 13 Dec 2019 07:46:30 GMT","content-type":"application/xml"}
2019-12-13T07:46:46.084Z 	 info: AccountDataStore:init() Refresh accounts from environment variable AZURITE_ACCOUNTS with value undefined
```
[debug.log](https://github.com/Azure/Azurite/files/3959704/debug.log)

### request_id
`1dc219ff-a189-4eff-84ce-f7104e63d92e`

**Comment by XiaoningLiu** [comment]
Should confirm. This likely to be incremental snpashot feature which is not supported.

**Comment by XiaoningLiu** [comment]
Fixed in #298
